src/wp-includes/class-wp-block.php

```diff
@@ -333,6 +333,64 @@ private function replace_html( string $block_content, string $attribute_name, $s
 		switch ( $block_type->attributes[ $attribute_name ]['source'] ) {
 			case 'html':
 			case 'rich-text':
+				if ( 'core/image' === $this->name && 'caption' === $attribute_name ) {
+					// Create private anonymous class until the HTML API provides `set_inner_html` method.
+					$bindings_processor = new class( $block_content, WP_HTML_Processor::CONSTRUCTOR_UNLOCK_CODE ) extends WP_HTML_Processor {
```

Comment:
given the way this works in PHP it could be useful to rename this something like `$bindings_processor_builder`

Reply:
Changed it as part of [this commit](https://github.com/WordPress/wordpress-develop/pull/6838/commits/825f78effbbe234ef70a3502eeaad78ee833436c).